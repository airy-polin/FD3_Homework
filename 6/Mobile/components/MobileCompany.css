.MobileCompany {
	width: 50rem;
	font-family: 'Open Sans', sans-serif;
	font-size: 14px;
}

.TableHeader {
	margin-bottom: 35px;
}
.TableHeader > .Buttons {
	display: flex;
	justify-content: center;
}
.TableHeader > .CompanyName {
	margin-left: 3px;
	font-weight: bold;
}

.TableFilter {
	display: flex;
}
table {
	margin: 15px 0;
	border: 2px solid #000;
	font-family: 'Open Sans', sans-serif;
	font-size: 14px;
	/* border-collapse: collapse; */
}
th, td {
	border: 1px solid #000;
}
th {
	background-color: lightcyan;
	padding: 10px;
}
td {
	padding: 0 10px;
}