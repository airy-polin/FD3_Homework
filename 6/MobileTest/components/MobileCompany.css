.MobileCompany {
	width: 50rem;
	font-family: 'Open Sans', sans-serif;
	font-size: 14px;
}

.TableFilter {
	display: flex;
}
table {
	margin: 15px 0;
	border: 2px solid #000;
	font-family: 'Open Sans', sans-serif;
	font-size: 14px;
	/* border-collapse: collapse; */
}
th, td {
	border: 1px solid #000;
}
th {
	background-color: lightcyan;
	padding: 10px;
}
td {
	padding: 0 10px;
}