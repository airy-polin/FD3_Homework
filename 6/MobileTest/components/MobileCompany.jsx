import React, { Fragment } from "react";
import PropTypes from 'prop-types';

import './MobileCompany.css';

import Button from "./Button.jsx";
import MobileClient from "./MobileClient.jsx";
import DataEditingForm from "./DataEditingForm.jsx";
import { myEvents } from "./events";

class MobileCompanyA1 extends React.PureComponent {
	static propTypes = {
		clients: PropTypes.arrayOf(
			PropTypes.shape({
				id: PropTypes.number.isRequired,
				fullName: PropTypes.shape({
					lastName: PropTypes.string.isRequired,
					firstName: PropTypes.string.isRequired,
					middleName: PropTypes.string.isRequired,
				}),
				balance: PropTypes.number.isRequired,
			})
		),
	};

	state = {
		clients: this.props.clients,
		filterMode: 0, // 0 - non-filtered list; 1 - active clients; 2 - blocked clients
		formMode: 0, // 0 - initial state (no mode); 1 - edit client mode; 2 - add new client mode
		editableClientID: null,
	};

	componentDidMount = () => {
		myEvents.addListener('EEditClient', this.editClient);
		myEvents.addListener('EDeleteClient', this.deleteClient);
		myEvents.addListener('ECancelChanges', this.cancelChanges);
		myEvents.addListener('ESaveChanges', this.saveChanges);
		myEvents.addListener('EAddNewClient', this.addNewClient);
	};

	componentWillUnmount = () => {
		myEvents.removeListener('EEditClient', this.editClient);
		myEvents.removeListener('EDeleteClient', this.deleteClient);
		myEvents.removeListener('ECancelChanges', this.cancelChanges);
		myEvents.removeListener('ESaveChanges', this.saveChanges);
		myEvents.removeListener('EAddNewClient', this.addNewClient);
	};

	filterActiveClients = () => {
		this.setState({
			filterMode: 1,
		});
	};

	filterBlockedClients = () => {
		this.setState({
			filterMode: 2,
		});
	};

	resetFilter = () => {
		this.setState({
			filterMode: 0,
		});
	};

	fitFilterHandler = (event) => {
		let clickedButton = event.target.innerText;

		if (clickedButton === 'Все') {
			this.resetFilter();
		}
		if (clickedButton === 'Активные') {
			this.filterActiveClients();
		}
		if (clickedButton === 'Заблокированные') {
			this.filterBlockedClients();
		}
	};

	addForm = () => {
		this.setState({
			formMode: 2,
		});
	};

	editClient = (id, newFormMode) => {
		this.setState({
			formMode: newFormMode,
			editableClientID: id,
		});
	};

	cancelChanges = () => {
		this.setState({
			formMode: 0,
		});
	};

	saveChanges = (editedClient) => {
		const allClients = this.state.clients;

		let editedClientIndex = allClients.findIndex(client => client.id === editedClient.id);
		const oldClient = this.state.clients[editedClientIndex];

		if (oldClient !== editedClient) {
			allClients.splice(editedClientIndex, 1, editedClient);
	
			this.setState({
				clients: allClients,
				formMode: 0,
			});
		} else {
			this.setState({
				formMode: 0,
			});
		}
	};

	deleteClient = (id, EO) => {
		console.log(EO.value);

		if (confirm(`Хотите удалить клиента ID ${id}?`)) {
			const allClients = this.state.clients.slice();
			const updatedClients = allClients.filter(client => client.id !== id);

			this.setState({
				clients: updatedClients,
			});
		}
	};

	addNewClient = (newClient) => {
		const allClients = this.state.clients.slice();
		allClients.push(newClient);

		this.setState({
			clients: allClients,
			formMode: 0,
		});
	};

	render() {
		console.log('MobileCompany rendered');

		let clientsToRender;
		const allClients = this.state.clients.slice();

		let newClientID = this.state.clients[this.state.clients.length - 1].id + 1;
		let editableClientIndex = allClients.findIndex(client => client.id === this.state.editableClientID);
		const editableClient = allClients[editableClientIndex];

		if (this.state.filterMode === 0) {
			clientsToRender = allClients;
		} else if (this.state.filterMode === 1) {
			clientsToRender = allClients.filter(client => client.balance >= 0);
		} else if (this.state.filterMode === 2) {
			clientsToRender = allClients.filter(client => client.balance < 0)
		}

		const clients = clientsToRender.map(client => 
			<MobileClient key={client.id} client={client} />
		);

		return (
			<Fragment>
				<div className='MobileCompany'>
					<div className='TableFilter'>
						<Button value='Все' handler={this.resetFilter} />
						<Button value='Активные' handler={this.filterActiveClients} />
						<Button value='Заблокированные' handler={this.filterBlockedClients} />
					</div>

					<div className='TableData'>
						<table>
							<thead>
								<tr className='TableHeader'>
									<th>Фамилия</th>
									<th>Имя</th>
									<th>Отчество</th>
									<th>Баланс</th>
									<th>Статус</th>
									<th>Редактировать</th>
									<th>Удалить</th>
								</tr>
							</thead>

							<tbody>
								{ clients }
							</tbody>
						</table>
					</div>

					<div className='TableEditor' onClick={this.addForm}>
						<Button value='Добавить клиента' />
					</div>
				</div>
				{
					(this.state.formMode === 1 || this.state.formMode === 2) &&
					<DataEditingForm
						formMode={this.state.formMode}
						newClientID={newClientID}
						client={editableClient}
					/>
				}
			</Fragment>
		);
	}
}

export default MobileCompanyA1;