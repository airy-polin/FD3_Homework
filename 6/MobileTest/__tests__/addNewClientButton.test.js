'use strict'

import React from 'react';
import renderer from 'react-test-renderer';
import ReactTestUtils from 'react-dom/test-utils';

import MobileCompany from '../components/MobileCompany';

const clients = [
	{ id: 111, fullName: { lastName: 'Иванов', firstName: 'Иван', middleName: 'Иванович' }, balance: 200 },
	{ id: 112, fullName: { lastName: 'Сидоров', firstName: 'Сидор', middleName: 'Сидорович' }, balance: 250 },
	{ id: 113, fullName: { lastName: 'Петров', firstName: 'Пётр', middleName: 'Петрович' }, balance: 180 },
	{ id: 114, fullName: { lastName: 'Григориев', firstName: 'Григорий', middleName: 'Григорьевич' }, balance: -220 },
];

// создаём тестовую версию компонента
const component = renderer.create(
	<MobileCompany clients={clients} />
);

// получаем снэпшот (HTML-снимок) компонента для сверки, что вёрстка не испортилась
let componentTree = component.toJSON();

beforeAll(() => {
	expect(componentTree).toMatchSnapshot();
});

describe('работа кнопки "Добавить клиента" для добавления нового клиента в список "доступных" клиентов компании', () => {
	// найдём в вёрстке компонента кнопку "Добавить клиента"
	const addNewClientBtn = component.root.find( el => el.props.value === 'Добавить клиента' );

	test('добавить нового клиента', () => {
		// нажимаем на кнопку "Добавить клиента"
		ReactTestUtils.Simulate.click(addNewClientBtn);

		// получаем изменённый снэпшот
		componentTree = component.toJSON();
		expect(componentTree).toMatchSnapshot('addNewClientButton.test.js.snap');
	});

	test('добавить еще одного нового клиента', () => {
		// нажимаем на кнопку "Добавить клиента" повторно
		ReactTestUtils.Simulate.click(addNewClientBtn);

		// получаем снова изменённый снэпшот
		componentTree = component.toJSON();
		expect(componentTree).toMatchSnapshot('addNewClientButton.test.js.snap');
	});
});