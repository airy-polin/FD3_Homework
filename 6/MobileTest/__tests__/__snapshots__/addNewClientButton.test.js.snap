// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div
  className="MobileCompany"
>
  <div
    className="TableFilter"
  >
    <div
      className="Button"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        Все
      </button>
    </div>
    <div
      className="Button"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        Активные
      </button>
    </div>
    <div
      className="Button"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        Заблокированные
      </button>
    </div>
  </div>
  <div
    className="TableData"
  >
    <table>
      <thead>
        <tr
          className="TableHeader"
        >
          <th>
            Фамилия
          </th>
          <th>
            Имя
          </th>
          <th>
            Отчество
          </th>
          <th>
            Баланс
          </th>
          <th>
            Статус
          </th>
          <th>
            Редактировать
          </th>
          <th>
            Удалить
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          className="TableLine"
        >
          <td>
            <span>
              Иванов
            </span>
          </td>
          <td>
            <span>
              Иван
            </span>
          </td>
          <td>
            <span>
              Иванович
            </span>
          </td>
          <td>
            <span>
              200
            </span>
          </td>
          <td
            style={
              Object {
                "backgroundColor": "lightgreen",
              }
            }
          >
            Active
          </td>
          <td
            onClick={[Function]}
          >
            <div
              className="Button"
            >
              <button
                type="button"
              >
                Редактировать
              </button>
            </div>
          </td>
          <td
            onClick={[Function]}
          >
            <div
              className="Button"
            >
              <button
                type="button"
              >
                Удалить
              </button>
            </div>
          </td>
        </tr>
        <tr
          className="TableLine"
        >
          <td>
            <span>
              Сидоров
            </span>
          </td>
          <td>
            <span>
              Сидор
            </span>
          </td>
          <td>
            <span>
              Сидорович
            </span>
          </td>
          <td>
            <span>
              250
            </span>
          </td>
          <td
            style={
              Object {
                "backgroundColor": "lightgreen",
              }
            }
          >
            Active
          </td>
          <td
            onClick={[Function]}
          >
            <div
              className="Button"
            >
              <button
                type="button"
              >
                Редактировать
              </button>
            </div>
          </td>
          <td
            onClick={[Function]}
          >
            <div
              className="Button"
            >
              <button
                type="button"
              >
                Удалить
              </button>
            </div>
          </td>
        </tr>
        <tr
          className="TableLine"
        >
          <td>
            <span>
              Петров
            </span>
          </td>
          <td>
            <span>
              Пётр
            </span>
          </td>
          <td>
            <span>
              Петрович
            </span>
          </td>
          <td>
            <span>
              180
            </span>
          </td>
          <td
            style={
              Object {
                "backgroundColor": "lightgreen",
              }
            }
          >
            Active
          </td>
          <td
            onClick={[Function]}
          >
            <div
              className="Button"
            >
              <button
                type="button"
              >
                Редактировать
              </button>
            </div>
          </td>
          <td
            onClick={[Function]}
          >
            <div
              className="Button"
            >
              <button
                type="button"
              >
                Удалить
              </button>
            </div>
          </td>
        </tr>
        <tr
          className="TableLine"
        >
          <td>
            <span>
              Григориев
            </span>
          </td>
          <td>
            <span>
              Григорий
            </span>
          </td>
          <td>
            <span>
              Григорьевич
            </span>
          </td>
          <td>
            <span>
              -220
            </span>
          </td>
          <td
            style={
              Object {
                "backgroundColor": "red",
              }
            }
          >
            Blocked
          </td>
          <td
            onClick={[Function]}
          >
            <div
              className="Button"
            >
              <button
                type="button"
              >
                Редактировать
              </button>
            </div>
          </td>
          <td
            onClick={[Function]}
          >
            <div
              className="Button"
            >
              <button
                type="button"
              >
                Удалить
              </button>
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
  <div
    className="TableEditor"
    onClick={[Function]}
  >
    <div
      className="Button"
    >
      <button
        type="button"
      >
        Добавить клиента
      </button>
    </div>
  </div>
</div>
`;

exports[`работа кнопки "Добавить клиента" для добавления нового клиента в список "доступных" клиентов компании добавить еще одного нового клиента 1`] = `
<div
  className="MobileCompany"
>
  <div
    className="TableFilter"
  >
    <div
      className="Button"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        Все
      </button>
    </div>
    <div
      className="Button"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        Активные
      </button>
    </div>
    <div
      className="Button"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        Заблокированные
      </button>
    </div>
  </div>
  <div
    className="TableData"
  >
    <table>
      <thead>
        <tr
          className="TableHeader"
        >
          <th>
            Фамилия
          </th>
          <th>
            Имя
          </th>
          <th>
            Отчество
          </th>
          <th>
            Баланс
          </th>
          <th>
            Статус
          </th>
          <th>
            Редактировать
          </th>
          <th>
            Удалить
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          className="TableLine"
        >
          <td>
            <span>
              Иванов
            </span>
          </td>
          <td>
            <span>
              Иван
            </span>
          </td>
          <td>
            <span>
              Иванович
            </span>
          </td>
          <td>
            <span>
              200
            </span>
          </td>
          <td
            style={
              Object {
                "backgroundColor": "lightgreen",
              }
            }
          >
            Active
          </td>
          <td
            onClick={[Function]}
          >
            <div
              className="Button"
            >
              <button
                type="button"
              >
                Редактировать
              </button>
            </div>
          </td>
          <td
            onClick={[Function]}
          >
            <div
              className="Button"
            >
              <button
                type="button"
              >
                Удалить
              </button>
            </div>
          </td>
        </tr>
        <tr
          className="TableLine"
        >
          <td>
            <span>
              Сидоров
            </span>
          </td>
          <td>
            <span>
              Сидор
            </span>
          </td>
          <td>
            <span>
              Сидорович
            </span>
          </td>
          <td>
            <span>
              250
            </span>
          </td>
          <td
            style={
              Object {
                "backgroundColor": "lightgreen",
              }
            }
          >
            Active
          </td>
          <td
            onClick={[Function]}
          >
            <div
              className="Button"
            >
              <button
                type="button"
              >
                Редактировать
              </button>
            </div>
          </td>
          <td
            onClick={[Function]}
          >
            <div
              className="Button"
            >
              <button
                type="button"
              >
                Удалить
              </button>
            </div>
          </td>
        </tr>
        <tr
          className="TableLine"
        >
          <td>
            <span>
              Петров
            </span>
          </td>
          <td>
            <span>
              Пётр
            </span>
          </td>
          <td>
            <span>
              Петрович
            </span>
          </td>
          <td>
            <span>
              180
            </span>
          </td>
          <td
            style={
              Object {
                "backgroundColor": "lightgreen",
              }
            }
          >
            Active
          </td>
          <td
            onClick={[Function]}
          >
            <div
              className="Button"
            >
              <button
                type="button"
              >
                Редактировать
              </button>
            </div>
          </td>
          <td
            onClick={[Function]}
          >
            <div
              className="Button"
            >
              <button
                type="button"
              >
                Удалить
              </button>
            </div>
          </td>
        </tr>
        <tr
          className="TableLine"
        >
          <td>
            <span>
              Григориев
            </span>
          </td>
          <td>
            <span>
              Григорий
            </span>
          </td>
          <td>
            <span>
              Григорьевич
            </span>
          </td>
          <td>
            <span>
              -220
            </span>
          </td>
          <td
            style={
              Object {
                "backgroundColor": "red",
              }
            }
          >
            Blocked
          </td>
          <td
            onClick={[Function]}
          >
            <div
              className="Button"
            >
              <button
                type="button"
              >
                Редактировать
              </button>
            </div>
          </td>
          <td
            onClick={[Function]}
          >
            <div
              className="Button"
            >
              <button
                type="button"
              >
                Удалить
              </button>
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
  <div
    className="TableEditor"
    onClick={[Function]}
  >
    <div
      className="Button"
    >
      <button
        type="button"
      >
        Добавить клиента
      </button>
    </div>
  </div>
</div>
`;

exports[`работа кнопки "Добавить клиента" для добавления нового клиента в список "доступных" клиентов компании добавить еще одного нового клиента: addNewClientButton.test.js.snap 1`] = `
<div
  className="MobileCompany"
>
  <div
    className="TableFilter"
  >
    <div
      className="Button"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        Все
      </button>
    </div>
    <div
      className="Button"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        Активные
      </button>
    </div>
    <div
      className="Button"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        Заблокированные
      </button>
    </div>
  </div>
  <div
    className="TableData"
  >
    <table>
      <thead>
        <tr
          className="TableHeader"
        >
          <th>
            Фамилия
          </th>
          <th>
            Имя
          </th>
          <th>
            Отчество
          </th>
          <th>
            Баланс
          </th>
          <th>
            Статус
          </th>
          <th>
            Редактировать
          </th>
          <th>
            Удалить
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          className="TableLine"
        >
          <td>
            <span>
              Иванов
            </span>
          </td>
          <td>
            <span>
              Иван
            </span>
          </td>
          <td>
            <span>
              Иванович
            </span>
          </td>
          <td>
            <span>
              200
            </span>
          </td>
          <td
            style={
              Object {
                "backgroundColor": "lightgreen",
              }
            }
          >
            Active
          </td>
          <td
            onClick={[Function]}
          >
            <div
              className="Button"
            >
              <button
                type="button"
              >
                Редактировать
              </button>
            </div>
          </td>
          <td
            onClick={[Function]}
          >
            <div
              className="Button"
            >
              <button
                type="button"
              >
                Удалить
              </button>
            </div>
          </td>
        </tr>
        <tr
          className="TableLine"
        >
          <td>
            <span>
              Сидоров
            </span>
          </td>
          <td>
            <span>
              Сидор
            </span>
          </td>
          <td>
            <span>
              Сидорович
            </span>
          </td>
          <td>
            <span>
              250
            </span>
          </td>
          <td
            style={
              Object {
                "backgroundColor": "lightgreen",
              }
            }
          >
            Active
          </td>
          <td
            onClick={[Function]}
          >
            <div
              className="Button"
            >
              <button
                type="button"
              >
                Редактировать
              </button>
            </div>
          </td>
          <td
            onClick={[Function]}
          >
            <div
              className="Button"
            >
              <button
                type="button"
              >
                Удалить
              </button>
            </div>
          </td>
        </tr>
        <tr
          className="TableLine"
        >
          <td>
            <span>
              Петров
            </span>
          </td>
          <td>
            <span>
              Пётр
            </span>
          </td>
          <td>
            <span>
              Петрович
            </span>
          </td>
          <td>
            <span>
              180
            </span>
          </td>
          <td
            style={
              Object {
                "backgroundColor": "lightgreen",
              }
            }
          >
            Active
          </td>
          <td
            onClick={[Function]}
          >
            <div
              className="Button"
            >
              <button
                type="button"
              >
                Редактировать
              </button>
            </div>
          </td>
          <td
            onClick={[Function]}
          >
            <div
              className="Button"
            >
              <button
                type="button"
              >
                Удалить
              </button>
            </div>
          </td>
        </tr>
        <tr
          className="TableLine"
        >
          <td>
            <span>
              Григориев
            </span>
          </td>
          <td>
            <span>
              Григорий
            </span>
          </td>
          <td>
            <span>
              Григорьевич
            </span>
          </td>
          <td>
            <span>
              -220
            </span>
          </td>
          <td
            style={
              Object {
                "backgroundColor": "red",
              }
            }
          >
            Blocked
          </td>
          <td
            onClick={[Function]}
          >
            <div
              className="Button"
            >
              <button
                type="button"
              >
                Редактировать
              </button>
            </div>
          </td>
          <td
            onClick={[Function]}
          >
            <div
              className="Button"
            >
              <button
                type="button"
              >
                Удалить
              </button>
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
  <div
    className="TableEditor"
    onClick={[Function]}
  >
    <div
      className="Button"
    >
      <button
        type="button"
      >
        Добавить клиента
      </button>
    </div>
  </div>
</div>
`;

exports[`работа кнопки "Добавить клиента" для добавления нового клиента в список "доступных" клиентов компании добавить нового клиента 1`] = `
<div
  className="MobileCompany"
>
  <div
    className="TableFilter"
  >
    <div
      className="Button"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        Все
      </button>
    </div>
    <div
      className="Button"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        Активные
      </button>
    </div>
    <div
      className="Button"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        Заблокированные
      </button>
    </div>
  </div>
  <div
    className="TableData"
  >
    <table>
      <thead>
        <tr
          className="TableHeader"
        >
          <th>
            Фамилия
          </th>
          <th>
            Имя
          </th>
          <th>
            Отчество
          </th>
          <th>
            Баланс
          </th>
          <th>
            Статус
          </th>
          <th>
            Редактировать
          </th>
          <th>
            Удалить
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          className="TableLine"
        >
          <td>
            <span>
              Иванов
            </span>
          </td>
          <td>
            <span>
              Иван
            </span>
          </td>
          <td>
            <span>
              Иванович
            </span>
          </td>
          <td>
            <span>
              200
            </span>
          </td>
          <td
            style={
              Object {
                "backgroundColor": "lightgreen",
              }
            }
          >
            Active
          </td>
          <td
            onClick={[Function]}
          >
            <div
              className="Button"
            >
              <button
                type="button"
              >
                Редактировать
              </button>
            </div>
          </td>
          <td
            onClick={[Function]}
          >
            <div
              className="Button"
            >
              <button
                type="button"
              >
                Удалить
              </button>
            </div>
          </td>
        </tr>
        <tr
          className="TableLine"
        >
          <td>
            <span>
              Сидоров
            </span>
          </td>
          <td>
            <span>
              Сидор
            </span>
          </td>
          <td>
            <span>
              Сидорович
            </span>
          </td>
          <td>
            <span>
              250
            </span>
          </td>
          <td
            style={
              Object {
                "backgroundColor": "lightgreen",
              }
            }
          >
            Active
          </td>
          <td
            onClick={[Function]}
          >
            <div
              className="Button"
            >
              <button
                type="button"
              >
                Редактировать
              </button>
            </div>
          </td>
          <td
            onClick={[Function]}
          >
            <div
              className="Button"
            >
              <button
                type="button"
              >
                Удалить
              </button>
            </div>
          </td>
        </tr>
        <tr
          className="TableLine"
        >
          <td>
            <span>
              Петров
            </span>
          </td>
          <td>
            <span>
              Пётр
            </span>
          </td>
          <td>
            <span>
              Петрович
            </span>
          </td>
          <td>
            <span>
              180
            </span>
          </td>
          <td
            style={
              Object {
                "backgroundColor": "lightgreen",
              }
            }
          >
            Active
          </td>
          <td
            onClick={[Function]}
          >
            <div
              className="Button"
            >
              <button
                type="button"
              >
                Редактировать
              </button>
            </div>
          </td>
          <td
            onClick={[Function]}
          >
            <div
              className="Button"
            >
              <button
                type="button"
              >
                Удалить
              </button>
            </div>
          </td>
        </tr>
        <tr
          className="TableLine"
        >
          <td>
            <span>
              Григориев
            </span>
          </td>
          <td>
            <span>
              Григорий
            </span>
          </td>
          <td>
            <span>
              Григорьевич
            </span>
          </td>
          <td>
            <span>
              -220
            </span>
          </td>
          <td
            style={
              Object {
                "backgroundColor": "red",
              }
            }
          >
            Blocked
          </td>
          <td
            onClick={[Function]}
          >
            <div
              className="Button"
            >
              <button
                type="button"
              >
                Редактировать
              </button>
            </div>
          </td>
          <td
            onClick={[Function]}
          >
            <div
              className="Button"
            >
              <button
                type="button"
              >
                Удалить
              </button>
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
  <div
    className="TableEditor"
    onClick={[Function]}
  >
    <div
      className="Button"
    >
      <button
        type="button"
      >
        Добавить клиента
      </button>
    </div>
  </div>
</div>
`;

exports[`работа кнопки "Добавить клиента" для добавления нового клиента в список "доступных" клиентов компании добавить нового клиента: addNewClientButton.test.js.snap 1`] = `
<div
  className="MobileCompany"
>
  <div
    className="TableFilter"
  >
    <div
      className="Button"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        Все
      </button>
    </div>
    <div
      className="Button"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        Активные
      </button>
    </div>
    <div
      className="Button"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        Заблокированные
      </button>
    </div>
  </div>
  <div
    className="TableData"
  >
    <table>
      <thead>
        <tr
          className="TableHeader"
        >
          <th>
            Фамилия
          </th>
          <th>
            Имя
          </th>
          <th>
            Отчество
          </th>
          <th>
            Баланс
          </th>
          <th>
            Статус
          </th>
          <th>
            Редактировать
          </th>
          <th>
            Удалить
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          className="TableLine"
        >
          <td>
            <span>
              Иванов
            </span>
          </td>
          <td>
            <span>
              Иван
            </span>
          </td>
          <td>
            <span>
              Иванович
            </span>
          </td>
          <td>
            <span>
              200
            </span>
          </td>
          <td
            style={
              Object {
                "backgroundColor": "lightgreen",
              }
            }
          >
            Active
          </td>
          <td
            onClick={[Function]}
          >
            <div
              className="Button"
            >
              <button
                type="button"
              >
                Редактировать
              </button>
            </div>
          </td>
          <td
            onClick={[Function]}
          >
            <div
              className="Button"
            >
              <button
                type="button"
              >
                Удалить
              </button>
            </div>
          </td>
        </tr>
        <tr
          className="TableLine"
        >
          <td>
            <span>
              Сидоров
            </span>
          </td>
          <td>
            <span>
              Сидор
            </span>
          </td>
          <td>
            <span>
              Сидорович
            </span>
          </td>
          <td>
            <span>
              250
            </span>
          </td>
          <td
            style={
              Object {
                "backgroundColor": "lightgreen",
              }
            }
          >
            Active
          </td>
          <td
            onClick={[Function]}
          >
            <div
              className="Button"
            >
              <button
                type="button"
              >
                Редактировать
              </button>
            </div>
          </td>
          <td
            onClick={[Function]}
          >
            <div
              className="Button"
            >
              <button
                type="button"
              >
                Удалить
              </button>
            </div>
          </td>
        </tr>
        <tr
          className="TableLine"
        >
          <td>
            <span>
              Петров
            </span>
          </td>
          <td>
            <span>
              Пётр
            </span>
          </td>
          <td>
            <span>
              Петрович
            </span>
          </td>
          <td>
            <span>
              180
            </span>
          </td>
          <td
            style={
              Object {
                "backgroundColor": "lightgreen",
              }
            }
          >
            Active
          </td>
          <td
            onClick={[Function]}
          >
            <div
              className="Button"
            >
              <button
                type="button"
              >
                Редактировать
              </button>
            </div>
          </td>
          <td
            onClick={[Function]}
          >
            <div
              className="Button"
            >
              <button
                type="button"
              >
                Удалить
              </button>
            </div>
          </td>
        </tr>
        <tr
          className="TableLine"
        >
          <td>
            <span>
              Григориев
            </span>
          </td>
          <td>
            <span>
              Григорий
            </span>
          </td>
          <td>
            <span>
              Григорьевич
            </span>
          </td>
          <td>
            <span>
              -220
            </span>
          </td>
          <td
            style={
              Object {
                "backgroundColor": "red",
              }
            }
          >
            Blocked
          </td>
          <td
            onClick={[Function]}
          >
            <div
              className="Button"
            >
              <button
                type="button"
              >
                Редактировать
              </button>
            </div>
          </td>
          <td
            onClick={[Function]}
          >
            <div
              className="Button"
            >
              <button
                type="button"
              >
                Удалить
              </button>
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
  <div
    className="TableEditor"
    onClick={[Function]}
  >
    <div
      className="Button"
    >
      <button
        type="button"
      >
        Добавить клиента
      </button>
    </div>
  </div>
</div>
`;
